src/components/hooks/useInstanceList.jsx

```diff
@@ -1,9 +1,9 @@
 import { useState, useEffect, useMemo } from 'react'
 import { useInfiniteQuery } from 'react-query'
-import { apiGetWidgetInstances } from '../../util/api'
+import { apiGetWidgetInstances, apiSearchWidgets } from '../../util/api'
 import { iconUrl } from '../../util/icon-url'
 
-export default function useInstanceList() {
+export default function useInstanceList(userOnly = true, query = "") {
```

Comment:
The more I think about this, the more I want this hook to be split into two different instance list hooks, because while there are some similarities, there are some key differences:

- They're hitting different API endpoints
- They're both requesting lists of widgets, but for different purposes. This especially affects the `widgets` query cache: one is a list of instances the user owns, the other is a list of instances the user may not expressly have access to. IMO, the `widgets` query cache should remain the list of widgets we own (in the My Widgets page) and the new hook would use a different query key.
- It makes references to the hooks elsewhere much cleaner as to what exactly the hook is being used for